InventoryUpdator.java

```diff
@@ -41,9 +47,193 @@ public void loadUserFile(){
 
     //will call most of the other methods to validate that an order is valid
     //This will be able to access the data base and save the changes to the file....../XDm
-    public void makeOrder(){
+    public void makeOrder() throws FileNotFoundException {
+        // Getting order information from user
+        OrderItem order = GetOrder();
 
+        // Processing order
+        processOrder(order);
     }
+
+
+    private void processOrder(OrderItem order) throws FileNotFoundException {
+        // Validating product id
+        if (!validateProductID(order.getProductId())){
+            System.out.println("Product does not exist in inventory.");
+
+            // Product id is invalid, adding order to failed orders
+            failedOrders.add(order);
+
+            return;
+        }
+
+        // Validating stock
+        if (!validateStock(order.getQuantity())){
+            if (item.getQuantity() == 0){
+                System.out.println("The product: "+ order.getProductId() + "is out of stock.");
+                return;
+            }
+            else{
+                System.out.println("Insufficient stock for product: " + order.getProductId() + " Only " + item.getQuantity() + " remaining");
+            }
+
+            // Not enough stock for product, adding order to failed orders
+            failedOrders.add(order);
+
+            return;
+        }
+
+        // Order is valid, updating quantity
+        int currentQuantity = item.getQuantity();
+        item.setQuantity(currentQuantity - order.getQuantity());
+
+        validOrders.add(order);
+
+        System.out.println();
+        System.out.println("Order completed successfully. New inventory value for product: ");
+        System.out.println("Product ID:        " + item.getProduct_id()
+                + "\nQuantity:          " + item.getQuantity()
+                + "\nWhole Sale Cost:   " + item.getWholesale_cost()
+                + "\nSale Price:        " + item.getSale_price()
+                + "\nSupplier ID:       " + item.getSupplier_id());
+
+        dataBase.saveFile();
+    }
+
+    private OrderItem GetOrder(){

```

Comment:
Copied most of @jcasta30's method from CustomerOrderDatatabase to get the order info from user. The only difference is that we don't add it to the customer order database we just return the OrderItem